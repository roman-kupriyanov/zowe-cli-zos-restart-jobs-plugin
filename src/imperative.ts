/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { IImperativeConfig } from "@zowe/imperative";

const config: IImperativeConfig = {
    commandModuleGlobs: ["**/cli/*/*.definition!(.d).*s"],
    pluginHealthCheck: __dirname + "/healthCheck.Handler",
    pluginSummary: "Zowe CLI restart z/OS jobs plug-in",
    rootCommandDescription: "Plugin, which allows to restart z/OS jobs",
    productDisplayName: "Zowe CLI z/OS Jobs Restart Plug-in",
    name: "zos-restart-jobs"
};

export = config;
