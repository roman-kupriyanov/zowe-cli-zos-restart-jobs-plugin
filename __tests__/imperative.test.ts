/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { IImperativeConfig } from "@zowe/imperative";

describe("imperative config", () => {

    // Will fail if imperative config object is changed. This is a sanity/protection check to ensure that any
    // changes to the configuration document are intended.
    it("should not have changed", () => {
        const config: IImperativeConfig = require("../src/imperative");
        expect(config).toBeDefined();
        expect(config.pluginHealthCheck).toContain("healthCheck.Handler");
        delete config.pluginHealthCheck;
        expect(config).toMatchSnapshot();
    });

});
