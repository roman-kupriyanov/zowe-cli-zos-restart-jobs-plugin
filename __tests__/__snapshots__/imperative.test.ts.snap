// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imperative config should not have changed 1`] = `
Object {
  "commandModuleGlobs": Array [
    "**/cli/*/*.definition!(.d).*s",
  ],
  "name": "zos-restart-jobs",
  "pluginSummary": "Zowe CLI restart z/OS jobs plug-in",
  "productDisplayName": "Zowe CLI z/OS Jobs Restart Plug-in",
  "rootCommandDescription": "Plugin, which allows to restart z/OS jobs",
}
`;
